src/utils/indices.rs

```diff
@@ -32,12 +32,13 @@ pub fn parent_indices(indices: &[usize]) -> Vec<usize> {
 }
 
 pub fn tree_depth(leaves_count: usize) -> usize {
-    if leaves_count == 1 {
-        1
-    } else {
-        let val = micromath::F32(leaves_count as f32);
-        val.log2().ceil().0 as usize
+    let mut current: usize = leaves_count;
```

Comment:
See https://github.com/antouhou/rs-merkle/pull/18/files for a more concise implementation.